### PR TITLE
Switch to JWS/JWK for NRTMv4 UNF sig

### PR DESCRIPTION
**Pull request by mxsasha**
Per https://github.com/mxsasha/nrtmv4/pull/40

### CODE REVIEW

File: irrd/mirroring/nrtm4/nrtm4_client.py [hunk 1]
```diff
@@ -143,14 +134,14 @@ def _retrieve_unf(self) -> Tuple[NRTM4UpdateNotificationFile, str]:
         )
         return unf, used_key
 
-    def _validate_unf_signature(self, unf_content: str, signature_b64: str) -> str:
+    def _deserialize_unf(self, unf_content: str) -> Tuple[bytes, str]:
         """
         Verify the Update Notification File signature,
         given the content (before JSON parsing) and a base64 signature.
```
Comment (mxsasha) [inline irrd/mirroring/nrtm4/nrtm4_client.py]:
```suggestion
        given the content (before JWS deserialize).
```

File: irrd/mirroring/nrtm4/nrtm4_client.py [hunk 2]
```diff
@@ -143,14 +134,14 @@ def _retrieve_unf(self) -> Tuple[NRTM4UpdateNotificationFile, str]:
...
         Returns the used key in base64 string.
```
Comment (mxsasha) [inline irrd/mirroring/nrtm4/nrtm4_client.py]:
```suggestion
        Returns the deserialized payload and used key in PEM string.
```

File: irrd/mirroring/nrtm4/nrtm4_client.py [hunk 3]
```diff
@@ -162,42 +153,45 @@ def _validate_unf_signature(self, unf_content: str, signature_b64: str) -> str:
...
-                f" {signature_b64}. The signature is valid for public key {config_key} set in the"
-                " nrtm4_client_initial_public_key setting, but that is only used for initial validation."
-                f" IRRD is currently expecting the public key {self.last_status.current_key}. If you want to"
-                " clear IRRDs key information and revert to nrtm4_client_initial_public_key, use the"
-                " 'irrdctl nrtmv4 client-clear-known-keys' command."
-            )
-            if self.last_status.next_key:
-                msg += f" or {self.last_status.next_key}"
-            raise NRTM4ClientError(msg)
+            if not key:  # pragma: no cover
```
Comment (mxsasha) [inline irrd/mirroring/nrtm4/nrtm4_client.py]: rename to key_pem or something

**Review by mxsasha — COMMENTED** [review]
TODO: docs, including an example of PEM in YAML

File: irrd/utils/crypto.py [hunk 4]
```diff
@@ -1,37 +1,49 @@
...
-from typing import Optional
+from typing import Optional, Union
 
-from cryptography.hazmat.primitives.asymmetric.ed25519 import (
-    Ed25519PrivateKey,
-    Ed25519PublicKey,
-)
+from joserfc import jws
+from joserfc.errors import JoseError
+from joserfc.rfc7515.model import CompactSignature
```
Comment (lepture) [inline irrd/utils/crypto.py:5]: Better with 

```suggestion
from joserfc.jws import CompactSignature
```

File: irrd/utils/crypto.py [hunk 5]
```diff
@@ -1,37 +1,49 @@
...
+from joserfc.rfc7518.ec_key import ECKey
```
Comment (lepture) [inline irrd/utils/crypto.py:6]: Better with

```suggestion
from joserfc.jwk import ECKey
```

Comment (mxsasha) [reply irrd/utils/crypto.py:6]: Thanks! https://github.com/irrdnet/irrd/pull/992

**Pull request merged**